### DISCUSSION TITLE
Syncing DB and Application versions?

### DISCUSSION THREAD

**Discussion by leo-amelyan** [grate-devs/grate#427] [Q&A]
In [documentation it says](https://erikbra.github.io/grate/getting-started/):

> We say version it using the same version as your code base by passing it a version number at runtime. We love versioning our applications based on source control revisions, and how sweet it is when you can pinpoint the exact state of your datatabase and the exact code in your application all to a given point in space and time in your repo!

Does this mean we sync the versions of my DB migrations and Application projects?

What if we require to run scripts that do not require changes to the Application?  Updates to the data but not schema. The Db Migration scripts should have incremented version and Application's version will stay the same.

**Comment by leo-amelyan** [comment]
@erikbra

**Comment by erikbra** [comment]
I don't have a definitive answer to this, of course. Your needs or deployment/versioning strategy needs to match _your_ workflow. In my previous experience, I tend to version the database as a separate deployable/component, and not syncronised with the application code versioning. But, of course you're free to choose whatever suits your needs best.
